galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt

```diff
@@ -1763,9 +1765,65 @@ abstract class VBlock(var form: VForm) : VConstants, DBContextHandler, ActionHan
   }
 
   protected fun fetchLookup(table: Int, currentField: VField) {
-    TODO()
-  }
 
+    val tab = object : Table(tables!![table]) {}
+    val columns = mutableListOf<Column<String>>()
+    val conditions = mutableListOf<Op<Boolean>>()
+
+    for (i in fields.indices) {
+      val f = fields[i]
+
+      if (f !== currentField && f.lookupColumn(table) != null && !f.isLookupKey(table)) {
+        f.setNull(activeRecord)
+      }
+    }
+
+    for (i in fields.indices) {
+      val column = fields[i].lookupColumn(table)
+      val col = Column<String>(tab, column!!, VarCharColumnType())
+
+
+      if (column != null) {
+        columns.add(col)
+      }
+      if (fields[i] == currentField || fields[i].isLookupKey(table)) {
+        val condition = fields[i].getSearchCondition()
+
+        val cond = object : Op<Boolean>() {
+          override fun toQueryBuilder(queryBuilder: QueryBuilder) {
+            TODO("Not yet implemented")
+          }
+        }
+
+        if (condition == null || !condition.startsWith("= ")) {
+          conditions.add(cond)
+        }
+      }
+    }
+
+    try {
+      transaction {
+        var t: Op<Boolean> = conditions[0]
+        conditions.minusElement(t).forEach {
+          t = t and it
+        }
+
+        val query = tab.slice(columns).select(t)
+
+        if (query.toList().isEmpty()) {
+          throw VExecFailedException(MessageCode.getMessage("VIS-00016",
+                  arrayOf<Any>(tables!![table])))
+        } else if (query.toList().isNotEmpty()) {
+
+          throw VExecFailedException(MessageCode.getMessage("VIS-00020",
+                  arrayOf<Any>(tables!![table])))
```

Comment:
There seem to be a missing part here : 
else {
        int     j = 0;

        for (int i = 0; i < fields.length; i++) {
          VField      f = fields[i];

          if (f.lookupColumn(table) != null) {
            f.setQuery(query, 1+j);
            j += 1;
          }
        }